### DISCUSSION TITLE
Some questions about improving the interface for yourself...

### DISCUSSION THREAD

**Discussion by Bunker56** [PhlexPlexico/G5V#146] [Q&A] [answered]
There are still questions, but these are already particulars. The question is - is it possible to slightly change the interface for different access rights? I am the only administrator so far and I need to make some changes in the regular user interface, that is, remove, for example, the menu item "Create a Match", the option "Add a card", etc., and it would be even better to make the user registration confirmed by the administrator, etc. Tell me in which direction to move, I will try to do what I need myself, so as not to bother you with questions once again...

**Comment by Bunker56** [comment]
To clarify the question, it is necessary to disable automatic registration and restrict entry only for admins and super admins... Where to see?

**Reply by PhlexPlexico** [reply]
Heya,

So the way the web panel is currently set up is to ensure that users are also able to setup and run their own matches. I don't plan on changing this anytime soon (I believe ebot web also allows for this). 

If you would like, you are more than free to make a fork and just adjust the routes to check if the user is an admin or super admin, and redirect as needed. I believe there is already a fork (though it may be dated) that already does this :)

**Reply by PhlexPlexico** [reply]
As for confirmation of registration, that is completely out of scope for this project and I most likely won't be looking into it. Per the API requirements, anyone is allowed to register and the API is more for tracking stats and allowing anyone to access their own profiles.

**Reply by PhlexPlexico** [reply]
And one last thing, if you REALLY wanted to, you could adjust the API on user registration and login to refer to the administrator list/super administrator list and only allow login/registration of those steam IDs. However, that is out of scope for this project.

**Reply by PhlexPlexico** [reply]
Also I was thinking more about this last night - I also am pretty sure I saw someone requesting this on a different website which is kind of funny. But anyways if you wanted to change the whole registration flow you would have to do something like this, also PRs would be more than welcome if this was configurable (i.e. `ADMIN_MODE` in the settings file and maybe appending to the `isLoggedIn` route to check if that flag is true) -

API:
- Create new setting in config file (boolean, this will be checked over all new code via an if statement)
- create a `pending_user` table with the migration scripts that mirrors the user table.
- during the registration/login route, check new variable. If this is true, check the admin/superadmin list to see if the steam id is there, if so, allow regular registration/login.
- if it's a new user and they are not a super admin, insert all their values into the pending user table. If it exists, then return a message saying their still pending acceptance. If not, add to table. This will prevent logins until they are in the user table.

Front-end:
As I said before, there are forks that exist of this repo that straight up remove routes or just check to see if an admin is viewing the page, and if not it redirects. I'm not really willing to accept PRs for that as there's no way to really determine how to build the front end, unless you make additional changes to the isLoggedIn call.

Hope that helps!

**Comment by Bunker56** [comment]
I will try. 
I have one, or rather two questions - an unauthorized user has an "All Teams" page for some reason empty (http://http://mmr.ydns.eu/teams) is this how it should be? And the second question is that the authorization procedure is in the API file /routes/users.js ?

**Reply by PhlexPlexico** [reply]
If they view the teams when not logged in, they will only be able to see the public teams.

The authorization you're looking for would be in the `auth.js` file. Under the callbacks, and local login and register.

**Reply by Bunker56** [reply]
Thanx!

**Comment by Bunker56** [comment, marked as answer]
I'm sorry about my previous questions. How do I disable the Steam authorization button and leave only the local login, otherwise I'm not a specialist in Vue programming...
I just want to disable Steam authorization and leave only the local login and temporarily block the availability of the local registration button in modal form.

**Reply by Bunker56** [reply]
Don't worry, I almost figured it out myself...
And with ADMIN_MODE, I need to think about it, a good idea.
